testsuite/tests/typing-gadts/gadthead.ml

```diff
@@ -25,6 +25,6 @@ let () = g M.x
 Line 2, characters 15-16:
 2 |   match x with I -> M.print I
                    ^
-Error: This pattern matches values of type "'a g"
+Error: This pattern matches values of type "_ g"
```

Comment:
This case might warrant a special case to not consider `_` as a user name.

Reply:
Good catch! I have addressed this in https://github.com/ocaml/ocaml/pull/13328/commits/f8fe04386f7df358e382265e2a6afb35d80566ec

This change is a bit ugly (in btype) and reverses the improvement in distant_errors.ml. I think I favor not doing it, but it's a close call. What do you think?

Reply:
I agree that the changes doesn't look good in practice, and is probably better to revert it.